server/src/main/java/org/elasticsearch/search/aggregations/metrics/AbstractTDigestPercentilesAggregator.java

```diff
@@ -75,18 +77,22 @@ public LeafBucketCollector getLeafCollector(LeafReaderContext ctx,
             return LeafBucketCollector.NO_OP_COLLECTOR;
         }
         final BigArrays bigArrays = context.bigArrays();
-        final SortedNumericDoubleValues values = valuesSource.doubleValues(ctx);
+        if (valuesSource instanceof ValuesSource.Histogram) {
```

Comment:
Is it a good idea to be doing an `instanceof` operation for every document here? Maybe the performance of `instanceof` is negligible these days but if not, given this is run in a very tight loop, maybe we should be choosing a "mode" in the constructor and then checking that enum/boolean each time instead?

Reply:
@colings86 I think this code is called once for every segment actually.

Reply:
oh, you are right, ignore this then :)